### PR TITLE
add files and images for comments

### PR DESCRIPTION
**Pull request by AnnPostnikova**
**ТЗ:** https://docs.google.com/document/d/1bz_xNaZw_kB6pxwB99OnRbHEukDpzb7i/edit

### CODE REVIEW

File: src/EventService.Data/EventCommentRepository.cs [hunk 1]
```diff
@@ -55,21 +56,32 @@ public async Task<bool> EditContentAsync(Guid commentId, JsonPatchDocument<DbEve
...
-    DbEventComment dbEventComment = await _provider.EventComments.FirstOrDefaultAsync(x => x.Id == commentId);
+    DbEventComment dbEventComment = await _provider.EventComments
+      .Include(x => x.Images)
+      .Include(x => x.Files)
+      .FirstOrDefaultAsync(x => x.Id == commentId);
 
     if (dbEventComment is null || request is null)
     {
-      return false;
+      return (false, null, null);
```
Comment (VLiamin) [inline src/EventService.Data/EventCommentRepository.cs]: default?

File: src/EventService.Data/EventRepository.cs [hunk 2]
```diff
@@ -157,9 +151,9 @@ public async Task<bool> EditAsync(Guid eventId, JsonPatchDocument<DbEvent> reque
 
     if (oldIsActive != newIsActive)
     {
-      _provider.EventsUsers.RemoveRange(_provider.EventsUsers.Where(x => x.EventId == eventId));
+      List<DbEventComment> comments = dbEvent.Comments
+        .Where(x => x.EventId == eventId && (x.Content != null)).ToList();
```
Comment (VLiamin) [inline src/EventService.Data/EventRepository.cs]: Do you need list?

File: src/EventService.Mappers/Db/DbEventCommentMapper.cs [hunk 3]
```diff
@@ -7,19 +8,30 @@ namespace LT.DigitalOffice.EventService.Mappers.Db;
 
 public class DbEventCommentMapper : IDbEventCommentMapper
 {
-  public DbEventComment Map(CreateEventCommentRequest request)
+  private readonly IDbImageMapper _imageMapper;
```
Comment (VLiamin) [inline src/EventService.Mappers/Db/DbEventCommentMapper.cs:11]: empty line

**Review by VLiamin — APPROVED** [review]

File: src/EventService.Data/FileRepository.cs [hunk 4]
```diff
@@ -5,79 +5,78 @@
...
+  {
+    if (filter is null)
+    {
+      return default;
+    }
+
+    IQueryable<DbFile> dbFilesQuery = _provider.Files.AsNoTracking();
+
+    return (
+      await dbFilesQuery.Where(file => file.EntityId == filter.EntityId)
```
Comment (SolovyovDmitry) [inline src/EventService.Data/FileRepository.cs]: move filtering to 59

File: src/EventService.Data/FileRepository.cs [hunk 5]
```diff
@@ -5,79 +5,78 @@
...
+        .Skip(filter.SkipCount).Take(filter.TakeCount).ToListAsync(),
+      await dbFilesQuery.CountAsync());
+  }
+
+  public Task<List<DbFile>> GetAsync(List<Guid> filesIds)
+  {
+    return _provider.Files.AsNoTracking().Where(x => filesIds.Contains(x.FileId)).ToListAsync();
+  }
+
+  public async Task<bool> CheckFilesAsync(Guid entityId, List<Guid> filesIds)
```
Comment (SolovyovDmitry) [inline src/EventService.Data/FileRepository.cs]: rename to describe actual checking. I can't say by the method name what is checking and what this method is supposed to return

File: src/EventService.Mappers/Db/DbEventCommentMapper.cs [hunk 6]
```diff
@@ -7,19 +8,31 @@ namespace LT.DigitalOffice.EventService.Mappers.Db;
...
+        Id = commentId,
         Content = request.Content,
         UserId = request.UserId,
         EventId = request.EventId,
         ParentId = request.ParentId,
         IsActive = true,
-        CreatedAtUtc = DateTime.UtcNow
+        CreatedAtUtc = DateTime.UtcNow,
+        Images = imagesIds?
+          .ConvertAll(imageId => _imageMapper.Map(imageId, commentId))
```
Comment (SolovyovDmitry) [inline src/EventService.Mappers/Db/DbEventCommentMapper.cs]: no need to split on 2 lines

File: src/EventService.Mappers/Db/DbEventCommentMapper.cs [hunk 7]
```diff
@@ -7,19 +8,31 @@ namespace LT.DigitalOffice.EventService.Mappers.Db;
...
+    IDbImageMapper imageMapper)
   {
+    _imageMapper = imageMapper;
+  }
+
+  public DbEventComment Map(CreateEventCommentRequest request, List<Guid> imagesIds)
+  {
+    Guid commentId = Guid.NewGuid();
+
     return request is null
```
Comment (SolovyovDmitry) [inline src/EventService.Mappers/Db/DbEventCommentMapper.cs]: deconstruct to if (null)-> return null so we can create guid only if it's really required

File: src/EventService.Validation/EventComment/CreateEventCommentRequestValidator.cs [hunk 8]
```diff
@@ -39,5 +42,10 @@ public class CreateEventCommentRequestValidator : AbstractValidator<CreateEventC
       .WithMessage("User id must not be empty.") 
       .MustAsync(async (userId, _) => await userService.CheckUsersExistenceAsync(new List<Guid>() { userId }, new List<string>()))
       .WithMessage("That user doesn't exist.");
+
+    When(x => !x.CommentImages.IsNullOrEmpty(),
+        () =>
```
Comment (SolovyovDmitry) [inline src/EventService.Validation/EventComment/CreateEventCommentRequestValidator.cs]: move to previous line and use {} after =>

**Review by SolovyovDmitry — APPROVED** [review]

**Review by ndoriomedov — APPROVED** [review]

**Pull request merged**